### PR TITLE
Add supports to set the OIDC username and groups prefix

### PR DESCRIPTION
**Pull request by omegas27**
Towards https://github.com/giantswarm/giantswarm/issues/10012

@giantswarm/team-ludacris I set that as a draft because I would need someones help to test it.

I will document it somewhere afterwards

### CODE REVIEW

File: service/controller/cluster.go [hunk 1]
```diff
@@ -118,10 +118,12 @@ func newClusterResourceSets(config ClusterConfig) ([]*controller.ResourceSet, er
 			NTPServers:         config.NTPServers,
 			ProjectName:        config.ProjectName,
 			OIDC: cloudconfig.OIDCConfig{
```
Comment (MarcelMue) [inline service/controller/cluster.go:120]: You will need to re-generate the golden files in these unit tests to make this build.

Comment (omegas27) [reply service/controller/cluster.go:122]: Sure :) 
How can I do that?

Comment (MarcelMue) [reply service/controller/cluster.go:122]: Actually that doesn't seem to be the issue - let me check. It builds for me locally

Comment (omegas27) [reply service/controller/cluster.go:122]: :+1:

Comment (MarcelMue) [reply service/controller/cluster.go:122]: @giantswarm/team-ludacris Can someone take a look here? I can not figure out why this build is failing.

Comment (tfussell) [reply service/controller/cluster.go:122]: Maybe the CircleCI runner is OOM? https://forum.golangbridge.org/t/go-build-exits-with-signal-killed/513

Comment (stone-z) [reply service/controller/cluster.go:122]: I had this repeatedly with kvm-operator too, intermittently. Retrying the build worked, not sure what the root cause is unless it is OOM or something Circle-side (my failures were right after a documented Circle outage so I guessed it was related)

Comment (omegas27) [reply service/controller/cluster.go:122]: It actually passed after a re-run

**Pull request merged**